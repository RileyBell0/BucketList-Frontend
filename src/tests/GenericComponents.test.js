import renderer from "react-test-renderer";
import {
  GenericInput,
  GenericButton,
  GenericImageInput,
  GenericInputParagraph,
  GenericParagraph,
  GenericDropdown,
} from "../components/GenericComponents";

it("Matches previous GenericInput snapshot", () => {
  const comp = renderer.create(
    <GenericInput
      type={"text"}
      title={"Test Name"}
      placeholder={"Test Placeholder"}
    />
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Matches previous GenericButton snapshot", () => {
  const comp = renderer.create(<GenericButton />);
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Matches previous GenericImageInput snapshot", () => {
  const comp = renderer.create(<GenericImageInput />);
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Matches previous GenericInputParagarph snapshot", () => {
  const tree = renderer
    .create(
      <GenericInputParagraph
        title="Test Title"
        defaultValue="Test Default Value"
      />,
      {
        createNodeMock: () => document.createElement("textarea"),
      }
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Matches previous GenericParagraph snapshot", () => {
  const comp = renderer.create(
    <GenericParagraph title={"Test Title"}>
      This is a test paragraph
    </GenericParagraph>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Matches previous GenericDropdown snapshot", () => {
  const comp = renderer.create(
    <GenericDropdown title={"Test Title"} options={["option 1", "option 2"]} />
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});
