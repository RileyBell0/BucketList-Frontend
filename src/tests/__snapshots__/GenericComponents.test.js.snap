// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches previous GenericButton snapshot 1`] = `
<div>
  <button
    className="generic-button undefined"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  />
</div>
`;

exports[`Matches previous GenericDropdown snapshot 1`] = `
<div>
  <p
    className="generic-field__name"
  >
    Test Title
  </p>
  <div
    className="generic-drop-down__container"
  >
    <select
      className="generic-drop-down"
    >
      <option>
        option 1
      </option>
      <option>
        option 2
      </option>
    </select>
  </div>
</div>
`;

exports[`Matches previous GenericImageInput snapshot 1`] = `
<div
  className="generic-image-input"
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  role="presentation"
  tabIndex={0}
>
  <input
    multiple={false}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <p>
    Click to Upload or Drag and Drop
  </p>
  <div
    className="generic-image-input__image-container"
  >
    <button
      className="generic-image-input__image-container__add-image"
      type="button"
    >
      +
    </button>
    <svg
      className="generic-image-input__image-container__image"
    >
      picture-icon.svg
    </svg>
  </div>
  <p
    className="generic-image-input__max-size"
  >
    5 MB limit
  </p>
  <br />
</div>
`;

exports[`Matches previous GenericInput snapshot 1`] = `
<div
  className="generic-input"
>
  <p
    className="generic-field__name"
  >
    Test Name
  </p>
  <input
    className="generic-input__input"
    placeholder="Test Placeholder"
    type="text"
  />
</div>
`;

exports[`Matches previous GenericInputParagarph snapshot 1`] = `
Array [
  <p
    className="generic-field__name"
  >
    Test Title
  </p>,
  <textarea
    className="generic-input-paragraph"
    defaultValue="Test Default Value"
    onChange={[Function]}
  />,
]
`;

exports[`Matches previous GenericParagraph snapshot 1`] = `
Array [
  <p
    className="generic-field__name"
  >
    Test Title
  </p>,
  <p
    className="generic-paragraph__container generic-paragraph undefined"
    onClick={[Function]}
  >
    This is a test paragraph
  </p>,
]
`;
